test.py

```diff
@@ -0,0 +1,74 @@
+import hydra
+from hydra.core.global_hydra import GlobalHydra
+from omegaconf import OmegaConf, DictConfig
+from torch_points3d.trainer import LitTrainer
+from torch_points3d.core.instantiator import HydraInstantiator, Instantiator
+from dataclasses import dataclass 
+from hydra.core.config_store import ConfigStore
+from typing import List, Any, Type
+from omegaconf import MISSING, OmegaConf
+from omegaconf._utils import is_structured_config
+
+OmegaConf.register_new_resolver("get_filename", lambda x: x.split("/")[-1])
+
+
+@dataclass 
+class TrainingDataConfig:
+    batch_size: int = 32
+    num_workers: int = 0
+    lr: float = MISSING
+
+# We seperate the dataset "cfg" from the actual dataset object
+# so that we can pass the "cfg" into the dataset constructors as a DictConfig
+# instead of as unwrapped parameters
+@dataclass
+class BaseDataConfig:
+    batch_size: int = 32
+    num_workers: int = 0
+    dataroot: str = "data"
+
+@dataclass
+class BaseDataset:
+    _target_: str
+    cfg: BaseDataConfig
+
+@dataclass
+class S3DISDataConfig(BaseDataConfig):
+    fold: int =  6
+
+@dataclass 
+class S3DISDataset(BaseDataset):
+    cfg: S3DISDataConfig
+
+@dataclass
+class Config:
+    dataset: Any
```

Comment:
I couldn't get the typing to work on this for some reason. You should be able to do `Type[BaseDataset]` but it gave me an error...